stactask/task.py

```diff
@@ -353,8 +367,20 @@ def upload_item_assets_to_s3(
         s3_client: Optional[s3] = None,
     ) -> Item:
         if self._upload:
+            if not self.payload.collection_options:
+                # preserve legacy behaviour
+                upload_options = self.payload.global_upload_options
+            else:
+                if item.collection_id is None:
+                    raise ValueError(
+                        f"Unable to get upload options for Item '{item.id}' because it "
+                        "does not have an assigned collection"
+                    )
+                upload_options = self.payload.get_collection_upload_options(
+                    item.collection_id
+                )
```

Comment:
Why not add a case to `self.payload.get_collection_upload_options` for a collection of `None` to fall back to the global `upload_options`? Actually, isn't this the current behavior as `self.get_collection_options(None)` will return `{}`? I think so, albeit the type annotations need to be updated to allow `None`.

In other words, I think throwing an error just because you don't have a collection assigned is a tad too restrictive. I think there are legitimate cases where an item may not have a collection. If we wanted to make requiring a collection opt-in elsewhere that's fine with me, but I am suspicious this is the correct place to do so even optionally.

Reply:
>Why not add a case to self.payload.get_collection_upload_options for a collection of None to fall back to the global upload_options? 

Agreed. Done.

>Actually, isn't this the current behavior as self.get_collection_options(None) will return {}? I think so, albeit the type annotations need to be updated to allow None.

Works, but does not pass mypy: `.get` expects a string, not a `str | None`.